File: server/models/Item.js
const { DataTypes } = require('sequelize');
const sequelize = require('./database');

// Player inventory items
const Item = sequelize.define('Item', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
    allowNull: false
  },
  userId: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'Users',
      key: 'id'
    }
  },
  itemId: {
    type: DataTypes.STRING,
    allowNull: false
  },
  name: {
    type: DataTypes.STRING,
    allowNull: false
  },
  type: {
    type: DataTypes.STRING,
    allowNull: false
  },
  // Store item details as JSON
  details: {
    type: DataTypes.JSON,
    allowNull: true
  },
  slot: {
    type: DataTypes.STRING,
    allowNull: true
  },
  stats: {
    type: DataTypes.JSON,
    allowNull: true
  },
  quality: {
    type: DataTypes.STRING,
    allowNull: true
  },
  equipped: {
    type: DataTypes.BOOLEAN,
    defaultValue: false
  }
  // Removed isActive field since it doesn't exist in the database table
});

// Export the model with its associated methods
module.exports = Item;

File: server-go/internal/models/item.go
package models

import "gorm.io/datatypes"

// Item 对应 Sequelize Item 模型

type Item struct {
	ID     string `gorm:"type:uuid;primaryKey;column:id"`
	UserID uint   `gorm:"column:user_id"`

	ItemID string `gorm:"column:item_id"`
	Name   string `gorm:"column:name"`
	Type   string `gorm:"column:type"`

	Details   datatypes.JSON `gorm:"column:details"`
	Slot      *string        `gorm:"column:slot"`    // 添加 Slot 字段，装备槽位
	Stats     datatypes.JSON `gorm:"column:stats"`   // 添加 Stats 字段，装备属性
	Quality   string         `gorm:"column:quality"` // 添加 Quality 字段，装备品质
	Equipped  bool           `gorm:"column:equipped"`
	EquipType *string        `gorm:"column:equip_type"` // 添加 EquipType 字段，装备类型
}

func (Item) TableName() string {
	return "items"
}
